src/geometry/point2d.rs

```diff
@@ -32,7 +32,15 @@ pub const CARDINALS_2D: [Point2D; 4] = [
     WEST_2D,
 ];
 
-
+pub fn cardinal_to_str(dir: &Point2D) -> Option<String> {
```

Comment:
we can probably pass point by value if we're dereferencing it anyways? not a big deal tho